source/dagger/gameplay/team_game/golem.cpp

```diff
@@ -93,66 +88,65 @@ void ancient_defenders::GolemBehaviorSystem::Run()
 			}
 			else if (golem_.currentAction == EAction::Attacking) {
 				AnimatorPlay(animation_, "ancient_defenders:golem:ATTACK_FRONT");
-				Engine::Registry().get<MageStats>(range_.target).health -= golem_.meleeDmg;
+				Engine::Registry().get<Health>(range_.target).currentHealth -= golem_.meleeDmg* Engine::DeltaTime();
 				golem_.currentAction = EAction::Moving; // Go back to moving after attacking
 			}
 		});
 
 }
 
-void ancient_defenders::GolemBehaviorSystem::OnEndOfFrame()
-{
-
-	auto view = Engine::Registry().view<MageStats>();
-
-	auto it = view.begin();
-	while (it != view.end()) {
-		auto& en = view.get<MageStats>(*it);
-		if (en.health <= 0.0f) {
-			Engine::Registry().destroy(*it);
-		}
-		it++;
-	}
-}
-
-Golem ancient_defenders::Golem::Get(Entity entity_)
+void ancient_defenders::EnemyBehaviorSystem::OnEndOfFrame()
 {
-	auto& reg = Engine::Registry();
-	auto& sprite = reg.get_or_emplace<Sprite>(entity_);
-	auto& pos = reg.get_or_emplace<Transform>(entity_);
-	auto& anim = reg.get_or_emplace<Animator>(entity_);
-	auto& gol = reg.get_or_emplace<Enemy>(entity_);
-	auto& col = reg.get_or_emplace<SimpleCollision>(entity_);
-	auto& roa = reg.get_or_emplace<RangeOfAttack>(entity_);
-
-	return Golem{ entity_, sprite, pos, anim, gol, col, roa };
 }
 
-Golem ancient_defenders::Golem::Create()
+Entity ancient_defenders::Golem::Create()
 {
 	auto& reg = Engine::Registry();
 	auto entity = reg.create();
-	auto gol = Golem::Get(entity);
+	auto& sprite = reg.emplace<Sprite>(entity);
+	auto& coordinates = reg.emplace<Transform>(entity);
+	auto& anim = reg.emplace<Animator>(entity);
+	auto& golem = reg.emplace<Enemy>(entity);
+	auto& hitbox = reg.emplace<SimpleCollision>(entity);
+	auto& range = reg.emplace<RangeOfAttack>(entity);
+	auto& health = reg.emplace<Health>(entity);
+
+	//AssignSprite(sprite, "spritesheets:golem-little-sheet:golem_stand_side:1"); // Skipped because it was creating issues with HP sprite, Animator will add the sprite
+	
+	auto start = WalkingPath::path.back();
 
-	AssignSprite(gol.sprite, "spritesheets:golem-little-sheet:golem_stand_side:1");
-	float ratio = gol.sprite.size.y / gol.sprite.size.x;
-	gol.sprite.scale = { 4,4 };
+    // Randomly create offset 
+    std::random_device dev;
+    std::mt19937 rng(dev());
 
-	auto start = WalkingPath::path.back();
+    std::uniform_int_distribution<std::mt19937::result_type> roll22(0, 22);
```

Comment:
22, 38 and other magic numbers can be made into constexpr values, that way u will get the same compilation time but better readability.